src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/roboticssupport/RoboticsNotificationServiceImpl.java

```diff
@@ -0,0 +1,169 @@
+            "Failed to send more time request to RPA");
+    }
+
+    @Override
+    public String rpaResponseNotifications(String referenceNumber) {
+        if (StringUtils.isEmpty(referenceNumber)) {
+            throw new BadRequestException("Reference number not supplied");
+        }
+        User user = userService.authenticateAnonymousCaseWorker();
+        String authorisation = user.getAuthorisation();
+        return resendRPA(referenceNumber, user.getAuthorisation(),
+            claim -> claim.getResponse().isPresent(),
+            claim -> defenceResponseNotificationService.notifyRobotics(
+                new DefendantResponseEvent(claim, authorisation)),
+            "Failed to send response to RPA");
+    }
+
+    @Override
+    public String rpaCCJNotifications(String referenceNumber) {
+        if (StringUtils.isEmpty(referenceNumber)) {
+            throw new BadRequestException("Reference number not supplied");
+        }
+        User user = userService.authenticateAnonymousCaseWorker();
+        String authorisation = user.getAuthorisation();
+        return resendRPA(referenceNumber, user.getAuthorisation(),
+            claim -> claim.getCountyCourtJudgment() != null,
+            claim -> ccjNotificationService.notifyRobotics(new CountyCourtJudgmentEvent(claim, authorisation)),
+            "Failed to send CCJ request to RPA");
+    }
+
+    @Override
+    public String rpaPIFNotifications(String referenceNumber) {
+        if (StringUtils.isEmpty(referenceNumber)) {
+            throw new BadRequestException("Reference number not supplied");
+        }
+        User user = userService.authenticateAnonymousCaseWorker();
+        String authorisation = user.getAuthorisation();
+        return resendRPA(referenceNumber, user.getAuthorisation(),
+            claim -> claim.getMoneyReceivedOn().isPresent(),
+            claim -> paidInFullNotificationService.notifyRobotics(new PaidInFullEvent(claim)),
+            "Failed to send paid-in-full statement to RPA");
+    }
+
+    private String resendRPA(
+        String reference,
+        String authorisation,
+        Predicate<Claim> precondition,
+        Consumer<Claim> consumer,
+        String errorMessage
+    ) {
+        String rpaState = RPA_STATE_FAILED;
+        Optional<Claim> claimOptional = claimService.getClaimByReference(reference, authorisation);
+        if (!claimOptional.isPresent()) {
+            rpaState = RPA_STATE_MISSING;
+            throw new BadRequestException(rpaState);
+        }
+        Claim claim = claimOptional.get();
+        if (!precondition.test(claim)) {
+            rpaState = RPA_STATE_INVALID;
+            throw new BadRequestException(rpaState);
+        }
+        try {
+            consumer.accept(claim);
+            return RPA_STATE_SUCCEEDED;
+        } catch (Exception ex) {
+            appInsightsExceptionLogger.warn(errorMessage, ex);
+            return rpaState + ": " + ex.getMessage();
```

Comment:
```suggestion
            return RPA_STATE_FAILED + ": " + ex.getMessage();
```
And remove rpaState variable